openapi3filter/req_resp_decoder.go

```diff
@@ -224,20 +224,66 @@ func decodeStyledParameter(param *openapi3.Parameter, input *RequestValidationIn
 
 func decodeValue(dec valueDecoder, param string, sm *openapi3.SerializationMethod, schema *openapi3.SchemaRef) (interface{}, error) {
 	var decodeFn func(param string, sm *openapi3.SerializationMethod, schema *openapi3.SchemaRef) (interface{}, error)
-	switch schema.Value.Type {
-	case "array":
-		decodeFn = func(param string, sm *openapi3.SerializationMethod, schema *openapi3.SchemaRef) (interface{}, error) {
-			return dec.DecodeArray(param, sm, schema)
+	if len(schema.Value.AnyOf) > 0 {
+		for _, sr := range schema.Value.AnyOf {
+			value, err := decodeValue(dec, param, sm, sr)
+			if value == nil && err == nil {
+				continue
+			}
+			return value, err
 		}
-	case "object":
-		decodeFn = func(param string, sm *openapi3.SerializationMethod, schema *openapi3.SchemaRef) (interface{}, error) {
-			return dec.DecodeObject(param, sm, schema)
+		return nil, nil
+	}
+	if len(schema.Value.AllOf) > 0 {
+		var value interface{}
+		var err error
+		for _, sr := range schema.Value.AllOf {
+			value, err = decodeValue(dec, param, sm, sr)
+			if value == nil || err != nil {
+				break
+			}
 		}
-	default:
-		decodeFn = dec.DecodePrimitive
+		return value, err
 	}
+	if len(schema.Value.OneOf) > 0 {
+		isMatched := 0
+		var value interface{}
+		for _, sr := range schema.Value.OneOf {
+			value, err := decodeValue(dec, param, sm, sr)
+			if err != nil {
+				return nil, err
+			}
+			if value != nil {
+				isMatched++
+			}
+		}
+		if isMatched == 1 {
+			return value, nil
+		}
+		if len(schema.Value.Required) == 0 {
```

Comment:
Ah you've mistaken JSON schema's `required` field with openapiv3's params `required` field.
Around https://github.com/getkin/kin-openapi/blob/02f5fafb329ca3c8f21b1f7cf1aeb80f8dd4c656/openapi3filter/req_resp_decoder.go#L222 an additional argument should be passed: `param.Required`.

Reply:
Yes  you are right, i have  mistaken about this.  and i have added this argu in decodeValue function.